File: java/json-transform/src/main/java/ai/bizone/jsontransform/functions/TransformerFunctionLong.java
package ai.bizone.jsontransform.functions;

import ai.bizone.jsontransform.functions.common.FunctionContext;
import ai.bizone.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionLong extends TransformerFunction {
    public TransformerFunctionLong() {
        super();
    }
    @Override
    public Object apply(FunctionContext context) {
        return context.getLong(null);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionLong.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { unwrapNumber } from "../JsonHelpers";

class TransformerFunctionLong extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    return unwrapNumber(await context.getLong(null));
  }
}

export default TransformerFunctionLong;
